### PR TITLE
remove avm use prompt, and insist on user installing a version if it's not yet installed

### LINKED ISSUE #1562: avm: make `avm use` return error if anchor version is not installed (instead of prompting)
this is done to make it avm more similar to nvm

### PR DESCRIPTION
**Pull request by SvenDowideit**
Closes #1562

replaces #1558

### CODE REVIEW

File: avm/src/lib.rs [hunk 1]
```diff
@@ -50,27 +49,21 @@ pub fn use_version(version: &Version) -> Result<()> {
...
+
+        return Err(anyhow!(
+            "You need to run 'avm install {}' to install it before using it.",
+            version
+        ));
     }
 
     let mut current_version_file = fs::File::create(current_version_file_path().as_path())?;
     current_version_file.write_all(version.to_string().as_bytes())?;
+    println!("Now using avm Version {}.", current_version()?);
```
Comment (tomlinton) [inline avm/src/lib.rs:66]:
```suggestion
    println!("Now using version {}.", current_version()?);
```

Comment (SvenDowideit) [reply avm/src/lib.rs:66]: oh man, i really shouldn't code while sick. I like the idea of telling the user what was installed

```
root@8d4fc19e1f61:/app# nvm use stable
Now using node v16.14.0 (npm v8.3.1)
root@8d4fc19e1f61:/app# 
root@8d4fc19e1f61:/app# 
root@8d4fc19e1f61:/app# nvm use v16.14.0
Now using node v16.14.0 (npm v8.3.1)
```

would it be reasonable to say `"Now using Anchor version {}"`?

Comment (SvenDowideit) [reply avm/src/lib.rs:66]:
```
sven@x1carbon:~/src/cryptoworkbench/anchor/avm$ cargo run --bin avm use 0.22.0
   Compiling avm v0.22.1 (/home/sven/src/cryptoworkbench/anchor/avm)
    Finished dev [unoptimized + debuginfo] target(s) in 3.37s
     Running `/home/sven/src/cryptoworkbench/anchor/target/debug/avm use 0.22.0`
Now using anchor version 0.22.0.
sven@x1carbon:~/src/cryptoworkbench/anchor/avm$ cargo run --bin avm use 0.22.1
    Finished dev [unoptimized + debuginfo] target(s) in 0.14s
     Running `/home/sven/src/cryptoworkbench/anchor/target/debug/avm use 0.22.1`
Now using anchor version 0.22.1.
sven@x1carbon:~/src/cryptoworkbench/anchor/avm$ cargo run --bin avm use 0.21.0
    Finished dev [unoptimized + debuginfo] target(s) in 0.09s
     Running `/home/sven/src/cryptoworkbench/anchor/target/debug/avm use 0.21.0`
Version 0.21.0 is not installed, staying on version 0.22.1.
Error: You need to run 'avm install 0.21.0' to install it before using it.
sven@x1carbon:~/src/cryptoworkbench/anchor/avm$ echo $?
1
```

**Comment by paul-schaaf** [comment]
LGTM. pls add a line to the breaking section of the changelog

**Comment by SvenDowideit** [comment]
```
root@4e25ea78ccf3:/# avm use 0.22.1
Error: Could not parse version file: unexpected end of input while parsing major version number
root@4e25ea78ccf3:/# avm use latest
Error: Could not parse version file: unexpected end of input while parsing major version number
```

I guess there's something else i need to learn about in the code :)

**Comment by SvenDowideit** [comment]
ok, added a fix for the initial condition

File: avm/src/lib.rs [hunk 2]
```diff
@@ -49,11 +49,13 @@ pub fn use_version(version: &Version) -> Result<()> {
     let installed_versions = read_installed_versions();
     // Make sure the requested version is installed
     if !installed_versions.contains(version) {
-        println!(
-            "Version {} is not installed, staying on version {}.",
-            version,
-            current_version()?
-        );
+        let current = current_version();
+        if !current.is_err() {
```
Comment (paul-schaaf) [inline avm/src/lib.rs:53]: you can simplify this with 
```rust
if let Ok(current) = current_version()? {
...
}
```

but I think we can still give some info even if there's no current version yet so a `match` on `current_version` sounds good where you still print `Version {} is not installed` if there's no current version

Comment (SvenDowideit) [reply avm/src/lib.rs]: yeah - more info helps those that are at my level and need to read it :+1:

**Review by paul-schaaf — APPROVED** [review]

**Comment by paul-schaaf** [comment]
thank you!

**Comment by SvenDowideit** [comment]
oops, sorry for the dependency :)

**Pull request merged**